include/swift/AST/Decl.h

```diff
@@ -3257,6 +3259,20 @@ class ClassDecl : public NominalTypeDecl {
     return getForeignClassKind() != ForeignKind::Normal;
   }
 
+  /// Returns true if the class has designated initializers that are not listed
+  /// in its members.
+  ///
+  /// This can occur, for example, if the class is an Objective-C class with
+  /// initializers that cannot be represented in Swift.
+  bool hasMissingDesignatedInitializers() const {
+    (void)getMembers();
```

Comment:
Can we `lookupDirect` on the name `init` instead, so we only deserialize initializers?